src/log/log.c

```diff
@@ -2246,8 +2246,10 @@ __wt_log_vprintf(WT_SESSION_IMPL *session, const char *fmt, va_list ap)
 		return (0);
 
 	va_copy(ap_copy, ap);
-	len = (size_t)vsnprintf(NULL, 0, fmt, ap_copy) + 1;
```

Comment:
We account for the terminating NUL when we get the initial length of the string to write into the log file and allocate a log record to hold it.  We use that length in computing the log size record and log messages assume a NUL-terminated string.